### PR TITLE
ref(github): Unify GitHub configuration btw social auth and SSO

### PR DESCRIPTION
**Pull request by BYK**
An extended retake of #16550 which collects all intersecting GitHub app configuration under the `github-login.` option name. Will follow up with onpremise and getsentry patches to remove the warnings and some clean up.

This patch should be totally safe and has identical behavior to the current system.

### CODE REVIEW

**Comment by BYK** [comment]
Ugh, hold on this will cause the same issue. Will update.

File: src/social_auth/backends/github.py [hunk 1]
```diff
@@ -23,14 +23,14 @@
 
 
 # GitHub configuration
-GITHUB_AUTHORIZATION_URL = "https://github.com/login/oauth/authorize"
-GITHUB_ACCESS_TOKEN_URL = "https://github.com/login/oauth/access_token"
-GITHUB_USER_DATA_URL = "https://api.github.com/user"
+GITHUB_BASE_DOMAIN = getattr(settings, "GITHUB_BASE_DOMAIN", "github.com")
```
Comment (scefali) [inline src/social_auth/backends/github.py:26]: @BYK when would `GITHUB_BASE_DOMAIN` change?

File: src/sentry/runner/initializer.py [hunk 2]
```diff
@@ -132,9 +132,25 @@ def get_asset_version(settings):
     "mail.use-tls": "EMAIL_USE_TLS",
     "mail.from": "SERVER_EMAIL",
     "mail.subject-prefix": "EMAIL_SUBJECT_PREFIX",
+    "github-login.client-id": "GITHUB_APP_ID",
```
Comment (scefali) [inline src/sentry/runner/initializer.py:135]: @BYK where are these new settings being used?

File: src/social_auth/backends/github.py (hunk 1 above)
Comment (BYK) [reply src/social_auth/backends/github.py:26]: When people are using self-hosted GitHub Enterprise

Comment (dcramer) [reply src/social_auth/backends/github.py:26]: are you sure this even works with GHE? i would probably avoid supporting something we historically haven't. folks can use normal SSO IMO.

File: src/sentry/runner/initializer.py (hunk 2 above)
Comment (BYK) [reply src/sentry/runner/initializer.py:135]: The way we do these migrations is: we introduce the options and deprecate the python settings creating a mapping between them. Then sometime in the future, we remove the python ones when we (and self-hosted users) migrated.

File: src/social_auth/backends/github.py (hunk 1 above)
Comment (scefali) [reply src/social_auth/backends/github.py:26]: @BYK I see but we shouldn't we be handling that at a different level (like a database row)? Unless we are adding functionality that won't exist in the SaaS product.

Comment (BYK) [reply src/social_auth/backends/github.py:26]: @dcramer @scefali - I mean I didn't really invent this. It was there for GitHub SSO and seems like these two things are sharing some code and config so I'm just trying to unify these.

**Comment by BYK** [comment]
Ready for review now :)

**Review by scefali — APPROVED** [review]
LGTM but maybe other people should look as well

**Comment by dashed** [comment]
@BYK @scefali This needs to be updated on `getsentry` fyi. 

Specifically `getsentry/conf/settings/defaults.py` and `getsentry/conf/settings/dev.py`.

**Comment by BYK** [comment]
@dashed - yup, aware. Just wanted to make sure this wasn't breaking anything like last time first.

**Pull request merged**